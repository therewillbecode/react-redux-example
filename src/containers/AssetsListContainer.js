import React, { PureComponent } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router";

import AssetsList from "../components/AssetsList";
import AssetFilter from "../components/AssetFilter";
import {
  filterAssetsByStatus,
  filterAssetsBySubstring
} from "../selectors/index";

export class AssetsListContainer extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { statusFilter: "all", searchQuery: "" };
    this.navigateToAsset = this.navigateToAsset.bind(this);
    this.updateStatusFilter = this.updateStatusFilter.bind(this);
    this.updateSearchQuery = this.updateSearchQuery.bind(this);
  }

  navigateToAsset(id) {
    this.props.history.push(`/asset/${id}`);
  }

  updateStatusFilter(e, { value }) {
    this.setState({ statusFilter: value });
  }

  updateSearchQuery(e, { value }) {
    this.setState({ searchQuery: value });
  }

  render() {
    const { assets } = this.props;
    const { statusFilter, searchQuery } = this.state;

    const filteredAssets = filterAssetsBySubstring(
      filterAssetsByStatus(assets, statusFilter),
      "name",
      searchQuery
    );

    return (
      <div>
        <AssetFilter
          handleChange={this.updateStatusFilter}
          updateSearchQuery={this.updateSearchQuery}
          searchQuery={searchQuery}
        />
        <AssetsList
          assets={filteredAssets}
          navigateToAsset={this.navigateToAsset}
        />
      </div>
    );
  }
}

const mapStateToProps = ({ assets }) => ({ assets });

export default withRouter(connect(mapStateToProps)(AssetsListContainer));
