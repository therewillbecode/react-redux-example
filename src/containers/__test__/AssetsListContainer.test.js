/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import sinon from "sinon";
import { shallow, mount } from "enzyme";

import { AssetsListContainer } from "../AssetsListContainer";
const assets = [
  {
    id: "kj48",
    name: "UAV",
    status: "dispatched",
    receivedTimestamp: "34245",
    dispatchedTimestamp: "5235",
    comment: "Lorem"
  }
];

describe("AssetsListContainer", () => {
  it("renders correctly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AssetsListContainer />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});

describe("Calls handler props", () => {
  let spy = null;

  beforeEach(() => {
    spy = new sinon.spy();
  });

  it("should navigate to correct route when navigateToAsset method is called", () => {
    const newStatusFilter = "dispatched";
    const id = "001";
    const history = { push: spy };
    const wrapper = mount(
      <AssetsListContainer assets={assets} history={history} />
    );
    wrapper.instance().navigateToAsset(id);
    expect(spy.firstCall.args).toEqual(["/asset/001"]);
  });

  it("should update state with new search query when updateSearchQuery method is called", () => {
    const newSearchQuery = "Foo";
    const wrapper = mount(<AssetsListContainer assets={assets} />);
    wrapper.instance().updateSearchQuery({}, { value: newSearchQuery });
    expect(wrapper.state().searchQuery).toEqual(newSearchQuery);
  });

  it("should update state with new status filter when updateStatusFilter method is called", () => {
    const newStatusFilter = "dispatched";
    const wrapper = mount(<AssetsListContainer assets={assets} />);
    wrapper.instance().updateStatusFilter({}, { value: newStatusFilter });
    expect(wrapper.state().statusFilter).toEqual(newStatusFilter);
  });
});
