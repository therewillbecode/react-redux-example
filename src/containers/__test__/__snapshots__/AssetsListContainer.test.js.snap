// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsListContainer renders correctly 1`] = `
<div>
  <AssetFilter
    handleChange={[Function]}
    searchQuery=""
    updateSearchQuery={[Function]}
  />
  <AssetsList
    assets={undefined}
    navigateToAsset={[Function]}
  />
</div>
`;
